### PR TITLE
[Merged by Bors] - vm: profile data for individual transactions

### PR DESCRIPTION
**Pull request by dshulyak**
```
go test ./genvm -run=xx -bench=BenchmarkTransactions
```

### CODE REVIEW

File: genvm/vm_test.go [hunk 1]
```diff
@@ -2046,6 +2046,94 @@ func FuzzParse(f *testing.F) {
...
+			require.NoError(b, err)
+			require.Empty(b, ineffective)
+			require.Equal(b, types.TransactionSuccess, txs[0].Status)
+			require.NoError(b, tt.Revert(lid.Sub(1)))
+		}
+	}
+
+	// benchmarks below will have overhead beside the transaction itself.
+	// they are useful mainly to collect execution profiles and make estimations based on them.
+	b.Run("singlesig/selfspawn", func(b *testing.B) {
```
Comment (countvonzero) [inline genvm/vm_test.go:2090]: i am curious why you omitted spawnTx in the benchmark.

**Review by countvonzero — APPROVED** [review]

**Comment by dshulyak** [comment]
12th Gen Intel(R) Core(TM) i9-12900KF and samsung nvme device
```
goos: linux
goarch: amd64
pkg: github.com/spacemeshos/go-spacemesh/genvm
cpu: 12th Gen Intel(R) Core(TM) i9-12900KF
BenchmarkTransactions/singlesig/selfspawn-24         	       1	7645998058 ns/op
BenchmarkTransactions/singlesig/spend-24             	       1	15646502276 ns/op
BenchmarkTransactions/multisig/selfspawn-24          	       1	16215524991 ns/op
BenchmarkTransactions/multisig/spend-24              	       1	31962386498 ns/op
BenchmarkTransactions/vesting/spawnvault-24          	       1	29394179143 ns/op
BenchmarkTransactions/vesting/drain-24               	       1	42539869994 ns/op
```

File: genvm/vm_test.go (hunk 1 above)
Comment (dshulyak) [reply genvm/vm_test.go:2090]: there is a spawnvault, i thought it is enough to understand relative difference, can add more

**Comment by dshulyak** [comment]
m2 with nvme

```
goos: darwin
goarch: arm64
pkg: github.com/spacemeshos/go-spacemesh/genvm
BenchmarkTransactions/singlesig/selfspawn-12         	       1	9325451542 ns/op
BenchmarkTransactions/singlesig/spend-12             	       1	19579077542 ns/op
BenchmarkTransactions/multisig/selfspawn-12          	       1	23834050292 ns/op
BenchmarkTransactions/multisig/spend-12              	       1	48487118958 ns/op
BenchmarkTransactions/vesting/spawnvault-12          	       1	41828753000 ns/op
BenchmarkTransactions/vesting/drain-12               	       1	57304248166 ns/op
PASS
ok  	github.com/spacemeshos/go-spacemesh/genvm	203.944s
```

**Comment by dshulyak** [comment]
```
goos: linux
goarch: amd64
pkg: github.com/spacemeshos/go-spacemesh/genvm
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkTransactions/singlesig/selfspawn-12         	       1	10263653890 ns/op

BenchmarkTransactions/singlesig/spend-12             	       1	22037335874 ns/op
BenchmarkTransactions/multisig/selfspawn-12          	       1	23515745483 ns/op
BenchmarkTransactions/multisig/spend-12              	       1	47050450892 ns/op
BenchmarkTransactions/vesting/spawnvault-12          	       1	42893327566 ns/op
BenchmarkTransactions/vesting/drain-12               	       1	61240700042 ns/op
PASS
ok  	github.com/spacemeshos/go-spacemesh/genvm	212.661s
```

**Comment by dshulyak** [comment]
```
goos: linux
goarch: amd64
pkg: github.com/spacemeshos/go-spacemesh/genvm
cpu: 12th Gen Intel(R) Core(TM) i9-12900KF
BenchmarkTransactions/singlesig/selfspawn-24         	       1	7649990394 ns/op
BenchmarkTransactions/singlesig/spawn-24             	       1	15947326421 ns/op
BenchmarkTransactions/singlesig/spend-24             	       1	15619175866 ns/op
BenchmarkTransactions/multisig/k=1/n=1/selfspawn-24  	       1	8357143639 ns/op
BenchmarkTransactions/multisig/k=1/n=1/spawn-24      	       1	17413478387 ns/op
BenchmarkTransactions/multisig/k=1/n=1/spend-24      	       1	20065873020 ns/op
BenchmarkTransactions/multisig/k=1/n=2/selfspawn-24  	       1	9073660651 ns/op
BenchmarkTransactions/multisig/k=1/n=2/spawn-24      	       1	18954120726 ns/op
BenchmarkTransactions/multisig/k=1/n=2/spend-24      	       1	20661216522 ns/op
BenchmarkTransactions/multisig/k=2/n=3/selfspawn-24  	       1	12173260689 ns/op
BenchmarkTransactions/multisig/k=2/n=3/spawn-24      	       1	24960579934 ns/op
BenchmarkTransactions/multisig/k=2/n=3/spend-24      	       1	26293129055 ns/op
BenchmarkTransactions/multisig/k=3/n=5/selfspawn-24  	       1	16381851927 ns/op
BenchmarkTransactions/multisig/k=3/n=5/spawn-24      	       1	32935687622 ns/op
BenchmarkTransactions/multisig/k=3/n=5/spend-24      	       1	33208412576 ns/op
BenchmarkTransactions/multisig/k=3/n=10/selfspawn-24 	       1	20247129612 ns/op
BenchmarkTransactions/multisig/k=3/n=10/spawn-24     	       1	40719311336 ns/op
BenchmarkTransactions/multisig/k=3/n=10/spend-24     	       1	37229650268 ns/op
BenchmarkTransactions/vesting/spawnvault-24          	       1	29604334293 ns/op
BenchmarkTransactions/vesting/drain-24               	       1	42866522953 ns/op
```

**Comment by dshulyak** [comment]
```
goos: linux
goarch: amd64
pkg: github.com/spacemeshos/go-spacemesh/genvm
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkTransactions/singlesig/selfspawn-12         	       1	11704525661 ns/op
BenchmarkTransactions/singlesig/spawn-12             	       1	26897418192 ns/op
BenchmarkTransactions/singlesig/spend-12             	       1	25901691919 ns/op
BenchmarkTransactions/multisig/k=1/n=1/selfspawn-12  	       1	14025322544 ns/op
BenchmarkTransactions/multisig/k=1/n=1/spawn-12      	       1	29957526320 ns/op
BenchmarkTransactions/multisig/k=1/n=1/spend-12      	       1	36050786089 ns/op
BenchmarkTransactions/multisig/k=1/n=2/selfspawn-12  	       1	15862689042 ns/op
BenchmarkTransactions/multisig/k=1/n=2/spawn-12      	       1	32954085388 ns/op
BenchmarkTransactions/multisig/k=1/n=2/spend-12      	       1	35949037523 ns/op
BenchmarkTransactions/multisig/k=2/n=3/selfspawn-12  	       1	20564815026 ns/op
BenchmarkTransactions/multisig/k=2/n=3/spawn-12      	       1	42913956191 ns/op
BenchmarkTransactions/multisig/k=2/n=3/spend-12      	       1	45770062349 ns/op
BenchmarkTransactions/multisig/k=3/n=5/selfspawn-12  	       1	26933099862 ns/op
BenchmarkTransactions/multisig/k=3/n=5/spawn-12      	       1	55813148721 ns/op
BenchmarkTransactions/multisig/k=3/n=5/spend-12      	       1	55530166262 ns/op
BenchmarkTransactions/multisig/k=3/n=7/selfspawn-12  	       1	29540656736 ns/op
BenchmarkTransactions/multisig/k=3/n=7/spawn-12      	       1	60641202153 ns/op
BenchmarkTransactions/multisig/k=3/n=7/spend-12      	       1	57765062053 ns/op
BenchmarkTransactions/vesting/spawnvault-12          	       1	49314965597 ns/op
BenchmarkTransactions/vesting/drain-12               	       1	70488608947 ns/op
```

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**